### PR TITLE
crossplane-cli: init at 1.14.3

### PR DESCRIPTION
**Pull request by selfuryon**
## Description of changes
Crossplane [released](https://docs.crossplane.io/latest/release-notes/1.14.0/) a [crossplane cli](https://docs.crossplane.io/latest/cli/) tool with 1.14.0 release version.

Nixpkgs already has a [crossplane](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/python-modules/crossplane/default.nix#L31) tool, so I decided to name it `crossplane-cli`.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix
Comment (eclairevoyant) [inline pkgs/top-level/all-packages.nix:4939]: pleasemove the package to `pkgs/by-name` and remove this line

File: pkgs/applications/networking/cluster/crossplane-cli/default.nix
Comment (eclairevoyant) [inline pkgs/applications/networking/cluster/crossplane-cli/default.nix:34]:
```suggestion
    description = "Utilities to make using Crossplane easier";
```

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
Thanks for the contribution! Some feedback below. Additionally, please split the maintainer list change into a separate commit (should go first)

File: pkgs/top-level/all-packages.nix
Comment (selfuryon) [reply pkgs/top-level/all-packages.nix]: Thanks for advice, fixed!

File: pkgs/applications/networking/cluster/crossplane-cli/default.nix
Comment (selfuryon) [reply pkgs/applications/networking/cluster/crossplane-cli/default.nix]: Done

File: pkgs/by-name/cr/crossplane-cli/package.nix
Comment (IogaMaster) [inline pkgs/by-name/cr/crossplane-cli/package.nix:13]: Please remove `refs/tags/`

**Review by IogaMaster — CHANGES_REQUESTED** [review]
Other than that looks great!
Welcome to nixpkgs and thank you for contributing! 😄

File: pkgs/by-name/cr/crossplane-cli/package.nix
Comment (eclairevoyant) [reply pkgs/by-name/cr/crossplane-cli/package.nix:13]: why?

Comment (IogaMaster) [reply pkgs/by-name/cr/crossplane-cli/package.nix:13]: I have never seen it before. It doesn't seem necessary.

Comment (eclairevoyant) [reply pkgs/by-name/cr/crossplane-cli/package.nix:13]: can we not do pointless nits? a quick search in this repo yields 2800+ results.

Comment (selfuryon) [reply pkgs/by-name/cr/crossplane-cli/package.nix]: Not a problem, fixed :)

**Review by IogaMaster — APPROVED** [review]
Looks great!

**Comment by selfuryon** [comment]
Do I need to do something for merging that? Ask somebody or do some additional actions?

**Review by WolfangAukang — APPROVED** [review]

**Comment by WolfangAukang** [comment]
> Do I need to do something for merging that? Ask somebody or do some additional actions?

The "best" option is to send your PR to [this thread](https://discourse.nixos.org/t/prs-already-reviewed/2617). Sometimes it gets merged, sometimes it gets ignored (unfortunately)

File: pkgs/by-name/cr/crossplane-cli/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/cr/crossplane-cli/package.nix:25]:
```suggestion
  subPackages = [ "cmd/crank" ];
```

File: pkgs/by-name/cr/crossplane-cli/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/cr/crossplane-cli/package.nix:37]:
```suggestion
    maintainers = with maintainers; [ selfuryon ];
```

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/by-name/cr/crossplane-cli/package.nix
Comment (selfuryon) [reply pkgs/by-name/cr/crossplane-cli/package.nix]: Fixed

File: pkgs/by-name/cr/crossplane-cli/package.nix
Comment (selfuryon) [reply pkgs/by-name/cr/crossplane-cli/package.nix]: Fixed

**Comment by selfuryon** [comment]
> @selfuryon I don't think it's necessary to change the name of the binary too, as an example see [`kubernetes-helm`](https://github.com/NixOS/nixpkgs/blob/63074ef47177bb7304cba1c7deca987739af0383/pkgs/applications/networking/cluster/helm/default.nix) which still uses the binary name `helm`.

It's the problem of naming:
* Crossplane already has `crossplane` application: it's original crossplane controller in Kubernetes pod
* Crossplane internally called this cli utility as `crank`, but in all [manuals](https://docs.crossplane.io/latest/cli/) reference it like `crossplane` or `crossplane cli`. For example, their [installation script](https://raw.githubusercontent.com/crossplane/crossplane/master/install.sh) rename `crank` to `crossplane`
* nixpkgs already has `crossplane` app, so I needed to specify crossplane somehow, so I decided to name it `crossplane-cli`.

But how do you suggest to name it? `crossplane-crank`?

File: pkgs/by-name/cr/crossplane-cli/package.nix
Comment (mjgallag) [inline pkgs/by-name/cr/crossplane-cli/package.nix:29]: @selfuryon my suggestion is to change the binary name to `crossplane` while leaving `pname` as `crossplane-cli` just as `pname` for Helm is `kubernetes-helm` to avoid conflict with existing `pname` `helm` but binary is still just `helm`. The [Crossplane CLI Command Reference](https://docs.crossplane.io/latest/cli/command-reference/) has binary as `crossplane` in its examples.

Comment (selfuryon) [reply pkgs/by-name/cr/crossplane-cli/package.nix]: Oh, got that! Done

**Review by mjgallag — APPROVED** [review]
@selfuryon I cannot merge as I'm not a maintainer but thank you for contributing this! I can confirm it works on my `x86_64-darwin` machine.

**Comment by emilytrau** [comment]
Thanks! Welcome to nixpkgs :)

**Pull request merged**
